homeassistant/components/tplink/light.py

```diff
@@ -269,6 +285,18 @@ def supported_features(self):
         """Flag supported features."""
         return self._light_features.supported_features
 
+    def _get_valid_temperature_range(self):
+        """Return the device-specific white temperature range (in Kelvin).
+
+        :return: White temperature range in Kelvin (minimum, maximum)
+        """
+        for obj, temp_range in TPLINK_KELVIN.items():
+            if re.match(obj, self.smartbulb.sys_info[LIGHT_SYSINFO_MODEL]):
```

Comment:
It's better to pass the already queried sysinfo to this method, as this would otherwise cause multiple I/O requests (until one matches) which are problematic with some devices.

Reply:
Fixed